### PR TITLE
Add some documentation for axisartist's ExtremeFinder, plus some cleanups.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

Let's try to figure out how this whole thing is constructed :)  At least
that PR is my understanding.

Minor refactorings as well.

~~Goes on top of https://github.com/matplotlib/matplotlib/pull/13878.~~

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/mpl_toolkits/axisartist/angle_helper.py [hunk 1]
```diff
@@ -335,26 +335,46 @@ def __call__(self, direction, factor, values):  # hour
...
+            *lon_cycle* or *lat_cycle*; in theory this can be any number but
+            the implementation actually assumes that it is 360 (if not None);
+            other values give nonsensical results.
+
+            This is done by "unwrapping" the transformed grid coordinates so
+            that jumps are less than a half-cycle; then normalizing the span to
+            no more than a full cycle.
+
+            For example, a range of (-360, -358) + (0, 2) is normalized to (-2,
+            2); a range of (5, 1000) is normalized to (5, 365).
```
[the remark this answers is not in the document]
Comment (anntzer) [reply lib/mpl_toolkits/axisartist/angle_helper.py]: Actually the example I gave had wrong values; edited that and reworded the thing in a hopefully clearer manner.

File: lib/mpl_toolkits/axisartist/angle_helper.py [hunk 2]
```diff
@@ -335,26 +335,49 @@ def __call__(self, direction, factor, values):  # hour
...
+
+            This is done by "unwrapping" the transformed grid coordinates so
+            that jumps are less than a half-cycle; then normalizing the span to
+            no more than a full cycle.
+
+            For example, if values are in the union of the [0, 2] and
+            [358, 360] intervals (typically, angles measured modulo 360), the
+            values in the second interval are normalized to [-2, 0] instead so
+            that the values now cover [-2, 2].  If values are in a range of
+            [5, 1000], this gets normalized to normalized to [5, 365].
```
Comment (QuLogic) [inline lib/mpl_toolkits/axisartist/angle_helper.py]:
```suggestion
            [5, 1000], this gets normalized to [5, 365].
```

File: lib/mpl_toolkits/axisartist/floating_axes.py [hunk 3]
```diff
@@ -143,16 +143,21 @@ def get_line(self, axes):
...
         """
-        get extreme values.
+        This subclass always returns the same bounding box.
 
-        x1, y1, x2, y2 in image coordinates (0-based)
-        nx, ny : number of division in each axis
+        Parameters
+        ----------
+        extremes : (float, float, float, float)
+            The bounding box always returned by this helper.
```
Comment (QuLogic) [inline lib/mpl_toolkits/axisartist/floating_axes.py]:
```suggestion
            The bounding box that will always returned by this helper.
```

**Comment by anntzer** [comment]
edited

**Comment by anntzer** [comment]
rebased

**Pull request merged**